### PR TITLE
Add parameter testing for powerscale, fix a few bugs

### PR DESCRIPTION
**Pull request by jooseppi-luna**
# Description
This PR adds four new powerscale e2e tests that install powerscale with different sample file values. It also adds a bunch of e2e testing infrastructure to support these tests and a couple minor bug fixes in pkg/drivers and pkg/utils:

- Add three new powerscale sample files to tests/e2e/testfiles that install powerscale with different parameters to more thoroughly test the driver installation
- Add corresponding storageclass files for the tests, the alt_vals_1 cr requires a different storage class
- Add three empty cert-secret files for the alt_vals_1 test case
- Add a script called check_parameters.sh to tests/e2e and corresponding .csv files in tests/e2e/testfiles that describes controller, node, and csm objects and uses stacked grep statements to check that the values of environment variables and other node parameters are what we are expecting them to be.
- Add a line to GetNode in pkg/drivers/commonconfig.go to save the environment variables for the node, before these were not getting set
- Add a debug statement in GetController for tolerations
- Fix a minor bug in pkg/utils/status.go where we use the Namespace instead of the Name to construct the names of pods, without this fix the name and the namespace had to be the same.
- Adds a few objects to the cleanup script in tests/e2e
- go.mod and go.sum updates
- Modifies the customTest field for e2e testing specified in tests/e2e/steps/step_common.go to accept multiple custom tests in the Run field.
- Correspondingly modifies the runCustomTest function in tests/e2e/steps/steps_def.go to handle Run as a list of strings rather than a single string
- Adds two new steps to the e2e testing at tests/e2e that set and clear node labels so that we can apply different labels to nodes in e2e testing -- the specific use case implemented here allows us to temporarily set all of the nodes to control-plane role by labeling them with the control-plane label and then removing the label when the tests are done.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/743 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration
Ran my e2e tests to ensure that they work. Ran unit tests to test code changes in pkg/drivers and pkg/utils.

### CODE REVIEW

File: pkg/utils/status.go [hunk 1]
```diff
@@ -162,7 +162,7 @@ func getDaemonSetStatus(ctx context.Context, instance *csmv1.ContainerStorageMod
 
 		failedCount := 0
 		podList := &corev1.PodList{}
-		label := instance.GetNamespace() + "-node"
+		label := instance.GetName() + "-node"
```
Comment (JacobGros) [inline pkg/utils/status.go:165]: Have other drivers tested this?
These changes will impact all drivers, right?

File: tests/e2e/steps/steps_def.go [hunk 2]
```diff
@@ -165,6 +165,26 @@ func (step *Step) validateDriverNotInstalled(res Resource, driverName string, cr
...
+	}
+
+	return nil
+}
+
+func (step *Step) removeNodeLabel(res Resource, label string) error {
+	if label == "control-plane" {
+		removeNodeLabel(label, "node-role.kubernetes.io/control-plane")
+	} else {
+		return fmt.Errorf("Setting worker node role to %s not supported, feel free to add support", label)
```
Comment (JacobGros) [inline tests/e2e/steps/steps_def.go:182]: should this be a different error message from setNodeLabel?

File: pkg/utils/status.go (hunk 1 above)
Comment (jooseppi-luna) [reply pkg/utils/status.go:165]: powerflex and powerscale -- yes, all drivers use this code

File: tests/e2e/steps/steps_def.go (hunk 2 above)
Comment (jooseppi-luna) [reply tests/e2e/steps/steps_def.go]: fixed

File: pkg/utils/status.go (hunk 1 above)
Comment (JacobGros) [reply pkg/utils/status.go:165]: Can we test/have other drivers test to make sure? 
Moose team has had a lot of operator defects lately- would like to not assume things will work without testing 
Once we fix e2e testing we won't need to explicitly consult other teams for each change

Comment (jooseppi-luna) [reply pkg/utils/status.go:165]: tested on other drivers as well, merging with changes that came in from PR this morning and then should be ready to go

**Review by abhi16394 — APPROVED** [review]

**Review by abhi16394 — APPROVED** [review]

**Review by JacobGros — APPROVED** [review]
LGTM

**Pull request merged**